src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationConfiguration.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Services.V3;
+
+namespace NuGet.Jobs.Catalog2Registration
+{
+    public class Catalog2RegistrationConfiguration : ICommitCollectorConfiguration
+    {
+        private static readonly int DefaultMaxConcurrentHivesPerId = Enum.GetValues(typeof(HiveType)).Length;
+
+        /// <summary>
+        /// The connection string used to connect to an Azure Blob Storage account. The connection string specifies
+        /// the account name, the endpoint suffix (e.g. Azure vs. Azure China), and authentication credential (e.g. storage
+        /// key).
+        /// </summary>
+        public string StorageConnectionString { get; set; }
+
+        /// <summary>
+        /// The blob storage container for the legacy hive (not gzipped, no SemVer 2.0.0 packages). This container is in
+        /// the Azure Blob Storage account specified in <see cref="StorageConnectionString"/>.
+        /// </summary>
+        public string LegacyStorageContainer { get; set; }
+
+        /// <summary>
+        /// The user-facing base URL for the legacy registration hive.
+        /// </summary>
+        public string LegacyBaseUrl { get; set; }
+
+        /// <summary>
+        /// The blob storage container for the gzipped hive (no SemVer 2.0.0 packages). This container is in the Azure
+        /// Blob Storage account specified in <see cref="StorageConnectionString"/>.
+        /// </summary>
+        public string GzippedStorageContainer { get; set; }
+
+        /// <summary>
+        /// The user-facing base URL for the gzipped registration hive.
+        /// </summary>
+        public string GzippedBaseUrl { get; set; }
+
+        /// <summary>
+        /// The blob storage container for the SemVer 2.0.0 hive (gzipped and SemVer 2.0.0 packages). This container is
+        /// in the Azure Blob Storage account specified in <see cref="StorageConnectionString"/>.
+        /// </summary>
+        public string SemVer2StorageContainer { get; set; }
+
+        /// <summary>
+        /// The user-facing base URL for the SemVer 2.0.0 registration hive.
+        /// </summary>
+        public string SemVer2BaseUrl { get; set; }
+
+        /// <summary>
+        /// Zero or more URL to the dependency cursor URLs. The registration collector will go no further in the
+        /// catalog than these cursors.
+        /// </summary>
+        public List<string> DependencyCursorUrls { get; set; }
+
+        /// <summary>
+        /// The catalog index URL to poll for package details and package deletes.
+        /// </summary>
+        public string Source { get; set; }
```

Comment:
Source may be misunderstood as the service index. What do you think of the name `CatalogIndexUrl`?

Reply:
I mimicked the [property name in Catalog2AzureSearch](https://github.com/NuGet/NuGet.Services.Metadata/blob/bc7b6b5f3f84728946785d2021487876b130eb24/src/NuGet.Services.AzureSearch/Catalog2AzureSearch/Catalog2AzureSearchConfiguration.cs#L14) and the other catalog reading jobs (e.g. [Catalog2Dnx](https://github.com/NuGet/NuGet.Services.Metadata/blob/bc7b6b5f3f84728946785d2021487876b130eb24/src/Ng/Jobs/Catalog2DnxJob.cs#L32)). I would rather remain consistent here.